discohorts/pipeline.py

```diff
@@ -97,13 +102,15 @@ def get_patient_to_work_dir(patients, work_dirs):
                         check_call(command)
```

Comment:
any potential gotchas about running the commands through `check_call`? e.g. where does the output go? I see that it raises an exception if the command fails and do you think it will require some cleanup procedure when that happens?.. 

Also: I think the `dry_run` should actually make a call to the pipeline script but with the `dry-run` option enabled. That would allow better testing of the full-stack?